src/hide_wav.cpp

```diff
@@ -2,6 +2,54 @@
 
 void stego::hide_wav() {
     //implement hiding in wav files
+      size_t pointer = 0;
+
+    output_buffer<<input_buffer.str().substr(pointer, 12);
+    pointer+=12;
+    
+    while(pointer<input_buffer.str().size()){
+    
+        string subchunk_id = input_buffer.str().substr(pointer, 4);     
+        pointer+=4;
+        int subchunk_size = *(int*)&input_buffer.str()[pointer];
+        pointer+=4;
+    
+        if(subchunk_id != "data"){
+            output_buffer<<subchunk_id;
+            output_buffer.write(&input_buffer.str()[pointer-8], 8);
+            output_buffer.write(&input_buffer.str()[pointer], subchunk_size);
+            pointer+=subchunk_size;
+        }
+        else{  
+            output_buffer << "data";
+            output_buffer.write((char*)&subchunk_size, 4);
+
+            char first_byte=0;
+            if (encryption_bit){
+                first_byte |= 0x01;
+            } 
+            if (compression_bit){
+                first_byte |= 0x02;
+            }  
+            output_buffer.put(first_byte);
```

Comment:
This section is a bit wrong, the first byte isn't read from the input and is instead defined as 0.
Read it from the input buffer/string then update accordingly.